src/box/iproto.cc

```diff
@@ -3038,36 +3089,70 @@ iproto_free(void)
 	evio_service_stop(&tx_binary);
 }
 
-int
-iproto_rmean_foreach(void *cb, void *cb_ctx)
+static int
+iproto_thread_rmean_foreach(struct rmean *rmean, void *cb, void *cb_ctx)
 {
-	for (size_t i = 0; i < RMEAN_NET_LAST; i++) {
+	int rc = 0;
+	for (size_t i = 0; i < rmean->stats_n; i++) {
+		int64_t mean = rmean_mean(rmean, i);
+		int64_t total = rmean_total(rmean, i);
+		if (((rmean_cb)cb)(rmean->stats[i].name, mean,
+				   total, cb_ctx) != 0)
+			rc = 1;
+	}
+	return rc;
+}
+
+static int
+iproto_rmean_foreach(const char **strings, size_t rmean_last_name,
+		     ptrdiff_t offset, void *cb, void *cb_ctx)
+{
+	for (size_t i = 0; i < rmean_last_name; i++) {
 		int64_t mean = 0;
 		int64_t total = 0;
-		for (int j = 0; j < iproto_threads_count; j++)  {
-			mean += rmean_mean(iproto_threads[j].rmean, i);
-			total += rmean_total(iproto_threads[j].rmean, i);
+		for (int j = 0; j < iproto_threads_count; j++) {
+			struct rmean **rmean =
+				(struct rmean **)
+				((char *)&iproto_threads[j] + offset);
+			assert(*rmean == iproto_threads[j].rmean ||
+			       *rmean == iproto_threads[j].tx.rmean);
+			mean += rmean_mean(*rmean, i);
+			total += rmean_total(*rmean, i);
 		}
-		int rc = ((rmean_cb)cb)(rmean_net_strings[i], mean,
-					total, cb_ctx);
+		int rc = ((rmean_cb)cb)(strings[i], mean, total, cb_ctx);
 		if (rc != 0)
 			return rc;
 	}
 	return 0;
 }
 
+int
+iproto_rmean_foreach(void *cb, void *cb_ctx)
+{
+	int rc;
+	if ((rc = iproto_rmean_foreach(rmean_net_strings,
```

Comment:
Please don't use name overload. Add _impl suffix to the helper function.

Reply:
fixed